---
title: "Introduction Project Lectoraat"
author: "Tiara Verweij"
date: "15-5-2021"
output: html_document
---

## Introduction

More and more biologists are struggling with digitisation. And by this I mean the combination of biology with data sciences. This is not something they have been trained in, but it is necessary in order to analyse certain datasets. Therefore, this poject looks at transcriptomics datasets (e.g. RNA sequencing datasets) and how these can be analysed without any programming experience. Therefore, the aim of our project is: To explore transcriptome datasets from the NCBI GEO database, without the use of programming. (1)

**making Data Downloading easy**

To analyse datasets, often a summarized experiment is made from one or more datasets. This is the input for a Shiny App, the shiny app then gives a clear overview with all the results in graphs and tables. In order to do this, a pipeline is created that downloads datasets from NCBI GEO by just filling in a GSE and SRA number. And after these datasets are downloaded, the data needs to be filtered so that it is ready to go into the Shiny App. (2)

**Using Salmon and DESEQ2**

To make the data ready for a summarized experiment, it has to be downloaded as Fastq. This is also the input for Salmon. Salmon is a tool for quantifying the expression of transcripts using RNA-seq data. (3)(4) After using Salmon we are going to do a Deseq2 analysis. We can use the output of Salmon for this. The Deseq2 analysis takes as input a table of raw counts. DESeq2 performs an internal normalization where geometric mean is calculated for each gene across all samples. The counts for a gene in each sample is then divided by this mean. The median of these ratios in a sample is the size factor for that sample. This procedure corrects for library size and RNA composition bias, which can arise for example when only a small number of genes are very highly expressed in one experiment condition but not in the other. (5)

**Shiny App**

After the deseq2 analysis, the data is in principle ready for the Shiny App.

**References**

(1)	https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE152439
(2)https://canvas.hu.nl/courses/20076/pages/project-3-lectoraat 
(3)	https://combine-lab.github.io/salmon/
(4) https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5600148/
(5)	https://chipster.csc.fi/manual/deseq2.html

